tests/extras/datasets/api/test_api_dataset.py

```diff
@@ -153,3 +153,37 @@ def test_exists_ok(self, requests_mocker, method):
         )
 
         assert api_data_set.exists()
+
+
+def test_successfully_load_with_content_response():
+    api_data_set = APIDataSet(
+        url="https://raw.githubusercontent.com/quantumblacklabs/kedro/develop/img/kedro_banner.png",
+        method="GET",
+        attribute="content",
+    )
+    content = api_data_set.load()
+    assert content[1:4] == b"PNG"  # part of PNG file signature
```

Comment:
this feels more like an E2E test rather than a unit test to me, if it's dependent on external factors like connecting to a real life URL and downloading data from it. Can you use `requests-mock` like in the other test cases and mock stuff to make it more `unit` test.

Maybe the whole `if attribute="json"` stuff can be moved out into a private function (look at how `MemoryDataSet`/`CachedDataSet` do it) to be easily unit-testable.

Reply:
Agree with @ZainPatelQB . All unit tests should run without internet connection.